app/lib/features/chat_ng/widgets/chat_input.dart

```diff
@@ -0,0 +1,617 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:acter/common/models/types.dart';
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/providers/keyboard_visbility_provider.dart';
+import 'package:acter/common/widgets/emoji_picker_widget.dart';
+import 'package:acter/common/widgets/frost_effect.dart';
+import 'package:acter/common/widgets/html_editor/html_editor.dart';
+import 'package:acter/features/attachments/actions/select_attachment.dart';
+import 'package:acter/features/chat/models/chat_input_state/chat_input_state.dart';
+import 'package:acter/features/chat/providers/chat_providers.dart';
+import 'package:acter/features/chat/utils.dart';
+import 'package:acter/features/chat/widgets/custom_input.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart' show MsgDraft;
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:emoji_picker_flutter/emoji_picker_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+import 'package:mime/mime.dart';
+import 'package:skeletonizer/skeletonizer.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:acter/common/extensions/options.dart';
+
+final _log = Logger('a3::chat::custom_input');
+
+final _allowEdit = StateProvider.family.autoDispose<bool, String>(
+  (ref, roomId) => ref.watch(
+    chatInputProvider
+        .select((state) => state.sendingState == SendingState.preparing),
+  ),
+);
+
+class ChatInput extends ConsumerWidget {
+  static const loadingKey = Key('chat-ng-loading');
+  static const noAccessKey = Key('chat-ng-no-access');
+
+  final String roomId;
+  final void Function(bool)? onTyping;
+
+  const ChatInput({super.key, required this.roomId, this.onTyping});
+
+  Widget _loadingWidget(BuildContext context) {
+    return Skeletonizer(
```

Comment:
The loading state of Skeletonizer actually shows the preview of the actual widget UI, so decoration is needed otherwise it'd go out of match in UI transition. Although we don't need expensive widgets here like ```AnimatedContainer``` etc.